django_password_validators/password_history/password_validation.py

```diff
@@ -42,26 +60,58 @@ def validate(self, password, user=None):
 
         if not self._user_ok(user):
             return
-
         for user_config in UserPasswordHistoryConfig.objects.filter(user=user):
             password_hash = user_config.make_password_hash(password)
             try:
-                PasswordHistory.objects.get(
-                    user_config=user_config,
-                    password=password_hash
-                )
-                raise ValidationError(
-                    _("You can not use a password that is already used in this application."),
-                    code='password_used'
-                )
+                if self.lookup_range == 0:
+                    raise PasswordHistory.DoesNotExist
+
+                current_user_passwords = PasswordHistory.objects.filter(
+                    user_config=user_config
+                ).order_by('date')
+
+                num_user_passwords = len(current_user_passwords)
+
+                if self.lookup_range >= num_user_passwords:
+                    PasswordHistory.objects.get(
+                        user_config=user_config,
+                        password=password_hash
+                    )
+                    raise self.validation_error
+
+                else:
+                    if any(
+                        entry.user_config == user_config and
+                        entry.password == password_hash
+                        for entry in current_user_passwords[
+                            num_user_passwords - self.lookup_range:
```

Comment:
@spokerman12, nit: can you add a wrapping parenthesis around this line so that the trailing `:` is more prominently visible and it is clear what it is used for? Just a suggestion :)

Reply:
@spokerman12, also with moving the deletion logic outside to a separate method that is called from elsewhere, do we even need the `if-else` block> Wouldn't just the code in  lines 76-80 above suffice?